security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -159,21 +154,6 @@ func (c *CitadelClient) buildConnection() (*grpc.ClientConn, error) {
 	return conn, nil
 }
 
-func (c *CitadelClient) reconnectIfNeeded() error {
-	if c.opts.ProvCert == "" || c.usingMtls.Load() {
```

Comment:
if we do this, can remove usingMtls as well I think

Reply:
good catch, i think we can remove it